lib/style/deprecations.ex

```diff
@@ -34,50 +34,72 @@ defmodule Styler.Style.Deprecations do
-         do: {f, fm, [path, lob, opts]}
-
-    # For ranges where `start > stop`, you need to explicitly include the step
-    # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
-    # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
-    defp style({{:., _, [{_, _, [module]}, :slice]} = f, funm, [enumerable, {:.., _, [_, _]} = range]})
-         when module in [:Enum, :String],
-         do: {f, funm, [enumerable, add_step_to_decreasing_range(range)]}
-
-    # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
-    defp style({:|>, m, [lhs, {{_, _, [{_, _, [:File]}, :stream!]} = f, fm, [{:__block__, _, [modes]} = opts, lob]}]})
-         when is_list(modes),
-         do: {:|>, m, [lhs, {f, fm, [lob, opts]}]}
-
-    # For ranges where `start > stop`, you need to explicitly include the step
-    # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
-    # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
-    defp style({:|>, m, [lhs, {{:., _, [{_, _, [mod]}, :slice]} = f, funm, [{:.., _, [_, _]} = range]}]})
-         when mod in [:Enum, :String],
-         do: {:|>, m, [lhs, {f, funm, [add_step_to_decreasing_range(range)]}]}
+  # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
+  defp style({{:., _, [{_, _, [:File]}, :stream!]} = f, fm, [path, {:__block__, _, [modes]} = opts, lob]})
+       when is_list(modes),
+       do: {f, fm, [path, lob, opts]}
+
+  # For ranges where `start > stop`, you need to explicitly include the step
+  # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
+  # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
+  defp style({{:., _, [{_, _, [module]}, :slice]} = f, funm, [enumerable, {:.., _, [_, _]} = range]})
+       when module in [:Enum, :String],
+       do: {f, funm, [enumerable, add_step_to_decreasing_range(range)]}
+
+  # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
+  defp style({:|>, m, [lhs, {{_, _, [{_, _, [:File]}, :stream!]} = f, fm, [{:__block__, _, [modes]} = opts, lob]}]})
+       when is_list(modes),
+       do: {:|>, m, [lhs, {f, fm, [lob, opts]}]}
+
+  # For ranges where `start > stop`, you need to explicitly include the step
+  # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
+  # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
+  defp style({:|>, m, [lhs, {{:., _, [{_, _, [mod]}, :slice]} = f, funm, [{:.., _, [_, _]} = range]}]})
+       when mod in [:Enum, :String],
+       do: {:|>, m, [lhs, {f, funm, [add_step_to_decreasing_range(range)]}]}
+
+  # ~R is deprecated in favor of ~r
+  defp style({:sigil_R, m, args}), do: {:sigil_r, m, args}
+
+  # For a decreasing range, we must use Date.range/3 instead of Date.range/2
+  defp style({{:., _, [{:__aliases__, _, [:Date]}, :range]} = funm, dm, [first, last]} = block) do
```

Comment:
In this [commit](https://github.com/adobe/elixir-styler/pull/123/commits/bdbb10f440651ee60c7a58b31e3e8251474c5d76) I decided to remove the `System.version` conditional because it should be safe to do these transformation asap, the reasoning behind that is:

* Explicit range step was introduced in v1.11
* `Date.range/3` was introduced in v1.12
* `sigil_r` was introduced in v1.0
* `:eof` in `IO.read` and `IO.binread` was introduced in v1.13